contracts/peripheral/YieldDirector.sol

```diff
@@ -0,0 +1,390 @@
+            to any address. Donors will be able to withdraw their principal
+            sOHM at any time. Donation recipients can also redeem accrued rebases at any time.
+ */
+contract YieldDirector is IYieldDirector, OlympusAccessControlled {
+    using SafeERC20 for IERC20;
+
+    uint256 private constant MAX_UINT256 = type(uint256).max;
+
+    address public immutable sOHM;
+
+    bool public depositDisabled;
+    bool public withdrawDisabled;
+    bool public redeemDisabled;
+
+    struct DonationInfo {
+        address recipient;
+        uint256 deposit; // Total non-agnostic amount deposited
+        uint256 agnosticDeposit; // Total agnostic amount deposited
+        uint256 carry; // Amount of sOHM accumulated over on deposit/withdraw
+        uint256 indexAtLastChange; // Index of last deposit/withdraw
+    }
+
+    struct RecipientInfo {
+        uint256 totalDebt; // Non-agnostic debt
+        uint256 carry; // Total non-agnostic value donating to recipient
+        uint256 agnosticDebt; // Total agnostic value of carry + debt
+        uint256 indexAtLastChange; // Index when agnostic value changed
+    }
+
+    mapping(address => DonationInfo[]) public donationInfo;
+    mapping(address => RecipientInfo) public recipientInfo;
+
+    event Deposited(address indexed donor_, address indexed recipient_, uint256 amount_);
+    event Withdrawn(address indexed donor_, address indexed recipient_, uint256 amount_);
+    event AllWithdrawn(address indexed donor_, uint256 indexed amount_);
+    event Donated(address indexed donor_, address indexed recipient_, uint256 amount_);
+    event Redeemed(address indexed recipient_, uint256 amount_);
+    event EmergencyShutdown(bool active_);
+
+    constructor (address sOhm_, address authority_)
+        OlympusAccessControlled(IOlympusAuthority(authority_))
+    {
+        require(sOhm_ != address(0), "Invalid address for sOHM");
+
+        sOHM = sOhm_;
+    }
+
+    /************************
+    * Donor Functions
+    ************************/
+
+    /**
+        @notice Deposit sOHM, records sender address and assign rebases to recipient
+        @param amount_ Amount of sOHM debt issued from donor to recipient
+        @param recipient_ Address to direct staking yield and vault shares to
+    */
+    function deposit(uint256 amount_, address recipient_) external override {
+        require(!depositDisabled, "Deposits currently disabled");
+        require(amount_ > 0, "Invalid deposit amount");
+        require(recipient_ != address(0), "Invalid recipient address");
+
+        IERC20(sOHM).safeTransferFrom(msg.sender, address(this), amount_);
+
+        uint256 index = IsOHM(sOHM).index();
+
+        // Record donors's issued debt to recipient address
+        DonationInfo[] storage donations = donationInfo[msg.sender];
+        uint256 recipientIndex = _getRecipientIndex(msg.sender, recipient_);
+
+        if(recipientIndex == MAX_UINT256) {
```

Comment:
do you mean unequal? if this equal max you would not be able to push another item in correct?

Reply:
Not sure i understand the purpose of this